mender-update/daemon/context.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-update/daemon/context.hpp>
+
+#include <common/log.hpp>
+
+namespace mender {
+namespace update {
+namespace daemon {
+
+namespace log = mender::common::log;
+namespace main_context = mender::update::context;
+
+const int STATE_DATA_VERSION = 2;
+
+ExpectedStateData ApiResponseJsonToStateData(const json::Json &json) {
+	StateData data;
+
+	expected::ExpectedString str = json.Get("id").and_then(json::ToString);
+	if (!str) {
+		return expected::unexpected(str.error().WithContext("Could not get deployment ID"));
+	}
+	data.update_info.id = str.value();
+
+	str = json.Get("artifact")
+			  .and_then([](const json::Json &json) { return json.Get("source"); })
+			  .and_then([](const json::Json &json) { return json.Get("uri"); })
+			  .and_then(json::ToString);
+	if (!str) {
+		return expected::unexpected(
+			str.error().WithContext("Could not get artifact URI for deployment"));
+	}
+	data.update_info.artifact.source.uri = str.value();
+	log::Debug("Artifact Download URL: " + data.update_info.artifact.source.uri);
+
+	str = json.Get("artifact")
+			  .and_then([](const json::Json &json) { return json.Get("source"); })
+			  .and_then([](const json::Json &json) { return json.Get("expire"); })
+			  .and_then(json::ToString);
+	if (str) {
+		data.update_info.artifact.source.expire = str.value();
+		// If it's not available, we don't care.
+	}
+
+	// For later: Update Control Maps should be handled here.
+
+	// Note: There is more information available in the response than we collect here, but we
+	// prefer to get the information from the artifact instead, since it is the authoritative
+	// source. And it's also signed, unlike the response.
+
+	return data;
+}
+
+const string kRollbackNotSupported = "rollback-not-supported";
```

Comment:
These strings and the helper functions using them should all be `static`, right?

Reply:
Hmm, yes. However, they are kind of "official" strings, since they are database keys. In the next commits I'm making, I've made more of these, but now available in the header, since they are needed elsewhere too.

I think instead of making them static, I'll add an extern declaration in the header, so that they are grouped together with the next strings I'm adding.